### PR TITLE
Create ConnectionDetails secret for PostgreSQL Server

### LINKED ISSUE #151: postgresql.azure.jet.crossplane.io should expose a secret with format username@hostname
postgresql.azure.jet.crossplane.io should expose a secret with format username@hostname so that it is compatible with postgresql.sql.crossplane.io which expects a secret in the mentioned format

### PR DESCRIPTION
**Pull request by ezgidemirel**
Signed-off-by: ezgidemirel <ezgidemirel91@gmail.com>

### Description of your changes
This change creates a ConnectionDetails k8s secret with the fields described in [here](postgresql-server-secret). 

Fixes #151 

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested
I've created a PostgreSQLServer with a public access firewall rule using provider-jet-azure. After that, using provider-sql, I created a Database instance with the ConnectionDetails secret of PostgreSQLServer.
```
NAME                                         READY   SYNCED   EXTERNAL-NAME   AGE
resourcegroup.azure.jet.crossplane.io/ezgi   True    True     ezgi            3d20h

NAME                                            READY   SYNCED   AGE
database.postgresql.sql.crossplane.io/ezgi-db   True    True     34s

NAME                                                             READY   SYNCED   EXTERNAL-NAME     AGE
server.dbforpostgresql.azure.jet.crossplane.io/ezgi-psqlserver   True    True     ezgi-psqlserver   91m

NAME                                                                 READY   SYNCED   EXTERNAL-NAME   AGE
firewallrule.dbforpostgresql.azure.jet.crossplane.io/ezgi-firewall   True    True     ezgi-firewall   11m
```

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: config/postgresql/config.go [hunk 1]
```diff
@@ -49,6 +53,14 @@ func Configure(p *config.Provider) {
 		r.ExternalName.GetExternalNameFn = common.GetNameFromFullyQualifiedID
 		// /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/mygroup1/providers/Microsoft.DBforPostgreSQL/servers/server1
 		r.ExternalName.GetIDFn = common.GetFullyQualifiedIDFn("Microsoft.DBforPostgreSQL", "servers", "name")
+		r.Sensitive.AdditionalConnectionDetailsFn = func(attr map[string]interface{}) (map[string][]byte, error) {
+			return map[string][]byte{
+				xpv1.ResourceCredentialsSecretUserKey:     []byte(fmt.Sprintf("%s@%s", attr["administrator_login"].(string), attr["name"].(string))),
```
Comment (ulucinar) [inline config/postgresql/config.go:58]: nit: 
```suggestion
				xpv1.ResourceCredentialsSecretUserKey:     []byte(fmt.Sprintf("%s@%s", attr["administrator_login"], attr["name"])),
```

File: config/postgresql/config.go [hunk 2]
```diff
@@ -19,7 +19,9 @@ package postgresql
 import (
 	"context"
 	"fmt"
+	"strconv"
 
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
```
Comment (ulucinar) [inline config/postgresql/config.go:24]: We can organize imports as follows:
<stdlib imports>
<3rd party imports>
<crossplane-runtime imports>
<local imports>

**Review by ulucinar — COMMENTED** [review]
Could you please extend PostgreSQL server's example manifest with a  `spec.writeConnectionSecretToRef`?

**Comment by ezgidemirel** [comment]
> Could you please extend PostgreSQL server's example manifest with a `spec.writeConnectionSecretToRef`? Could you please also make sure that Terraform native provider does not allow configuration of CIDR whitelists?

I have extended the example manifest. I also check TF registry for firewall rules. Looks like they defined a separate [resource](https://registry.terraform.io/providers/hashicorp/azurerm/2.78.0/docs/resources/postgresql_firewall_rule) for configuring whitelisting rules.

**Review by ulucinar — APPROVED** [review]

**Comment by ulucinar** [comment]
Thank you @ezgidemirel.

**Review by jbw976 — COMMENTED** [review]
a little late to the game here, but is there something similar we should do for mysql and MSSQL in this provider too? 🤔

**Comment by ulucinar** [comment]
> a little late to the game here, but is there something similar we should do for mysql and MSSQL in this provider too? 🤔

Hi @jbw976,
We also discussed this with @ezgidemirel while reviewing this PR but to move fast, we had decided to open separate issue for it.

**Comment by jbw976** [comment]
totally reasonable approach @ulucinar! glad to get this PR in quickly, and we can do more support for the other DB types soon 💪

**Pull request merged**